integration-tests/src/omnipool_liquidity_mining.rs

```diff
@@ -0,0 +1,614 @@
+// This file is part of HydraDX-node.
+
+// Copyright (C) 2020-2023  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(test)]
+use crate::polkadot_test_net::*;
+
+use frame_support::{assert_noop, assert_ok};
+use warehouse_liquidity_mining::{
+	DepositData, GlobalFarmData, GlobalFarmId, Instance1, LoyaltyCurve, YieldFarmData, YieldFarmEntry,
+};
+
+use orml_traits::MultiCurrency;
+use polkadot_primitives::v2::BlockNumber;
+use primitives::AssetId;
+use sp_runtime::{
+	traits::{One, Zero},
+	FixedPointNumber, FixedU128, Permill, Perquintill,
+};
+use xcm_emulator::TestExt;
+
+use hydradx_runtime::{Balance, Origin};
+use pretty_assertions::assert_eq;
+
+#[macro_export]
+macro_rules! assert_nft_owner {
+	( $coll:expr, $item: expr, $acc:expr ) => {{
+		assert_eq!(hydradx_runtime::Uniques::owner($coll, $item).unwrap(), $acc);
+	}};
+}
+
+#[test]
+fn create_global_farm_should_work_when_origin_is_root() {
+	TestNet::reset();
+
+	Hydra::execute_with(|| {
+		let total_rewards: Balance = 1_000_000 * UNITS;
+		let planned_yielding_periods: BlockNumber = 1_000_000;
+		let blocks_per_period: BlockNumber = 10;
+		let reward_currency = HDX;
+		let owner = Treasury::account_id();
+		let yield_per_period = Perquintill::from_parts(570_776_255_707);
+		let min_deposit = 1_000;
+		//3[LRNA] = 1[HDX]
+		let lrna_price_adjustment = FixedU128::checked_from_rational(1, 3).unwrap();
+
+		assert_ok!(hydradx_runtime::Balances::set_balance(
```

Comment:
as this is used in many places, how about making a helper method in the root somewhere?

it is quite common across our integration tests, so it would be a handy helper method I feel